src/test/kotlin/app/batch/HBaseReaderTest.kt

```diff
@@ -111,6 +111,67 @@ class HBaseReaderTest {
             expected.toString(), actual.toString())
     }
 
+    @Test
+    fun testInnerType() {
```

Comment:
We need a test for when the @type field is empty and one for when the @type field doesn't exist.

Reply:
Done.